src/spartan/polys/eq.rs

```diff
@@ -14,8 +14,9 @@ use rayon::prelude::{IndexedParallelIterator, IntoParallelRefMutIterator, Parall
 /// This polynomial evaluates to 1 if every component $x_i$ equals its corresponding $e_i$, and 0 otherwise.
 ///
 /// For instance, for e = 6 (with a binary representation of 0b110), the vector r would be [1, 1, 0].
+#[derive(Debug)]
 pub struct EqPolynomial<Scalar: PrimeField> {
-  r: Vec<Scalar>,
+  pub(in crate::spartan::polys) r: Vec<Scalar>,
```

Comment:
Is there reason to make `r` visible?

Reply:
`MaskedEq` has an access to those scalars that doesn't exactly match an evaluation. And you can look at `evals_from_points` which has allowed us to save a ton of allocations:
https://github.com/lurk-lab/arecibo/pull/131/commits/13773b93086c878d8772356766308fd7b4747279